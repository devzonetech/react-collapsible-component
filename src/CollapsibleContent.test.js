import React from "react";
import { CollapsibleContent } from "./index";
import renderer from "react-test-renderer";

test("CollapsibleContent render and class property, children type test", () => {
    const component = renderer.create(
        <CollapsibleContent
            className="additionalClassForContent"
            isExpanded={true}
        >
            <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                eiusmod tempor
            </p>
        </CollapsibleContent>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(tree.props.className).toEqual(
        " panel-accordion additionalClassForContent"
    );
    expect(tree.children[0].type).toEqual("p");

    const expandStyle = { display: "block" };
    expect(tree.props.style).toEqual(expandStyle);
});

test("CollapsibleContent expand test", () => {
    const component = renderer.create(
        <CollapsibleContent
            className="additionalClassForContent"
            isExpanded={false}
        >
            <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                eiusmod tempor
            </p>
        </CollapsibleContent>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    console.log(tree.props);

    const expandStyle = null;
    expect(tree.props.style).toEqual(expandStyle);
});
