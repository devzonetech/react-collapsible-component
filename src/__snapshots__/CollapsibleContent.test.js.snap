// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleContent expand test 1`] = `
<div
  className=" panel-accordion additionalClassForContent"
  style={null}
>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
  </p>
</div>
`;

exports[`CollapsibleContent render and class property, children type test 1`] = `
<div
  className=" panel-accordion additionalClassForContent"
  style={
    Object {
      "display": "block",
    }
  }
>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
  </p>
</div>
`;
